app/controllers/api/databases.php

```diff
@@ -2885,12 +2885,16 @@ function updateAttribute(
 
         $collection = Authorization::skip(fn() => $dbForProject->getDocument('database_' . $database->getInternalId(), $collectionId));
 
-        if ($collection->isEmpty() || !$collection->getAttribute('enabled')) {
-            if (!($mode === APP_MODE_ADMIN && Auth::isPrivilegedUser(Authorization::getRoles()))) {
-                throw new Exception(Exception::COLLECTION_NOT_FOUND);
+        if (!($mode === APP_MODE_ADMIN && Auth::isPrivilegedUser(Authorization::getRoles()))) {
+            if (!$collection->getAttribute('documentSecurity', false)) {
+                $collection = $dbForProject->getDocument('database_' . $database->getInternalId(), $collectionId);
             }
         }
```

Comment:
This actually changes the behavior a bit. Do we want to return 404 if the collection exists, but the user doesn't have access to it? I think the way it was before, list documents would return 200 with 0 documents.

Reply:
We're going to move forward with returning 404